pkgs/applications/window-managers/notion/default.nix

```diff
@@ -0,0 +1,34 @@
+{
+  enableXft ? true, libXft ? null,
+  patches ? [],
+  fallbackFont ? "Sans",
+  stdenv, fetchurl,
+  lua, gettext, groff,
+  pkgconfig, busybox,
+  x11, libXinerama, libXrandr, libX11
+}:
+
+assert enableXft -> libXft != null && fallbackFont != null;
+
+stdenv.mkDerivation {
+  name     = "notion";
+  version  = "3-2015061300";
+  meta = with stdenv.lib; {
+    description = "Tiling tabbed window manager, follow-on to the ion window manager";
+    homepage = http://notion.sourceforge.net;
+    platforms = platforms.linux;
+    license   = licenses.notion_lgpl;
+    maintainers = [maintainers.jfb];
+  };
+  src = fetchurl {
+    url = https://github.com/raboof/notion/archive/3-2015061300.tar.gz;
+    sha256 = "3c9d9f35a9fb0d17c263b76fe28f7a1a4a05b7d6140545524cc1effd98c5c305";
+  };
+
+  patches = patches ++ stdenv.lib.optional enableXft [./notion-xft_nixos.diff];
```

Comment:
You don't like copy-pasting, do you? :) `lib.optional` already returns a list. Just `patches = patches ++ stdenv.lib.optional enableXft ./notion-xft_nixos.diff;` as suggested before should work.